### ISSUE TITLE
How frequently do validator registrations need to be published?

### ISSUE THREAD

**Issue by ralexstokes** [ethereum/builder-specs#24]
The spec currently says validators should submit their registrations once per epoch.

However, it seems like the mev-boost component has the responsibility to keep registrations it learns about and upstream them to connected relays in which case the validator client may be posting data every ~6 mins for no additional gain.

I'm opening this issue to seek feedback on expectations around this process and if we can reduce the frequency I'm happy to change the spec to reflect that.

Based on my current understanding, the validator client only needs to call the registration API when they boot or if any of their registration data changes (which practically may be synonymous with boot if clients only allow changes to config during validator client process boot).

**Comment by realbigsean** [comment]
Something to think about is if we spec out once per epoch registrations, will the relay be flooded with hundreds of thousands of registrations on the first slot of every epoch?

**Comment by metachris** [comment]
Yep, that's correct, and we are also just mapping out the infra to handle this. I don't think there's a good way around this, since we want relays coming online later to catch up with the latest registrations.

The service receiving the validator registrations needs to be able to handle large bursts of registrations, and validate them without impacting the other systems. The service will need a list of validators in memory to do a first validity check of the messages before verifying the signatures.

We're going to load test with 1M-10M of requests.

**Comment by ralexstokes** [comment]
we could add some randomized jitter to when exactly in the epoch (e.g. add a random number of fractional slots from 0-SLOTS_PER_EPOCH) the registration is rebroadcast to avoid any hotspots like this
